### PR TITLE
Add a more debug friendly bot

### PR DESCRIPTION
**Pull request by Ephememory**
Surrounded by #DEBUG preprocessor.

Adds a bot that makes debugging a bit easier than the standard s&box MimicBot. This bot can be toggled to mimic or wander vaguely randomly. Includes functionality for killing individual bots and kicking all bots. (You can already kick individual clients by name in s&box.) 

I originally included functionality for making all the bots attack and for aimbotting the player from my gamemode. I could add this back if you want.

### CODE REVIEW

File: code/Other/DebugBot.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+{
+	/// <summary>
+	/// The Player pawn this bot controls.
+	/// </summary>
+	public Player Pawn;
+
+	/// <summary>
+	/// The host Player who spawned this bot.
+	/// </summary>
+	public Player Target;
```
Comment (matekdev) [inline code/Other/DebugBot.cs:18]: `Owner`? or maybe `Spawner`?

File: code/Other/DebugBot.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		Wander = false;
+		Mimic = false;
+	}
+
+	[ConCmd.Admin( "bot_kill", Help = "Kills a bot by name if provided, else kills all bots." )]
+	public static void BotKill( string name = "" )
+	{
+		if ( string.IsNullOrEmpty( name ) )
+		{
+			foreach ( var b in All.ToArray() )
```
Comment (matekdev) [inline code/Other/DebugBot.cs:52]: Is this `Client.All`?

File: code/Other/DebugBot.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	{
+		foreach ( var b in All.ToArray() )
+			b.Client.Kick();
+	}
+
+	[ConCmd.Admin( "bot_add" )]
+	public static void AddBot()
+	{
+		var pawn = ConsoleSystem.Caller.Pawn;
+		if ( pawn is not Player ply )
```
Comment (matekdev) [inline code/Other/DebugBot.cs:80]: stick to `player` since we use that everwhere else.

File: code/Other/DebugBot.cs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+			return;
+
+		var b = new DebugBot();
```
Comment (matekdev) [inline code/Other/DebugBot.cs:83]: `bot`

File: code/Other/DebugBot.cs (hunk 2 above)
Comment (Ephememory) [reply code/Other/DebugBot.cs:52]: It's Bot.All but it can be simplified to All

Comment (Ephememory) [reply code/Other/DebugBot.cs:52]: //
		// Summary:
		//     Contains all bots on the server.
		public static IReadOnlyList<Bot> All => _all.AsReadOnly();

**Pull request merged**